packages/rocketchat-reactions/setReaction.js

```diff
@@ -39,12 +39,21 @@ Meteor.methods({
 			return false;
 		}
 
-		if (message.reactions && message.reactions[reaction] && message.reactions[reaction].usernames.indexOf(user.username) !== -1) {
+		const userAlreadyReacted = message.reactions && message.reactions[reaction] && message.reactions[reaction].usernames.indexOf(user.username) !== -1;
+		const removeUserReaction = () => {
 			message.reactions[reaction].usernames.splice(message.reactions[reaction].usernames.indexOf(user.username), 1);
-
+		};
+		const removeMessageReactionIfNeed = () => {
 			if (message.reactions[reaction].usernames.length === 0) {
 				delete message.reactions[reaction];
 			}
+		};
+		if (userAlreadyReacted) {
```

Comment:
why not just do something like?
```javascript
if(userAlreadyReacted === shouldReact) {
	return;
}
```

Reply:
I think we should not simply return if they are the same in this case, because if we do not throw an error when we try to react in a message already reacted, the REST API will end up returning success, without actually having performed the react action. What do you think?

Reply:
I'm not sure... but this change aims allow set the reaction instead toggle. if you want react but you already reacted, I think you got it, its not a ERROR...

Reply:
I think there's no need to throw an error if the end result is going to be the same as when the method is successful.

Reply:
We'd need to change quite a few REST things, then. We have more methods that error with "The current status is the same as it would be set to"

Reply:
It's something that is worth looking into, then. We should get everybody's opinion on which approach is better and then standardize every method.